### PR TITLE
Update helm chart and add test to ci

### PR DESCRIPTION
**Pull request by willmostly**
## Description

Add Helm chart testing and fix https://github.com/trinodb/trino-gateway/issues/401

## Additional context and related issues

Add linting and a test install to CI. Bring Helm chart up to date with Airlift style configuration.

## Release notes

( ) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
(x) Release notes are required, with the following suggested text:

Improve Helm chart reliability

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -37,3 +37,24 @@ jobs:
...
+          python-version: '3.10'
+          check-latest: true
+      - name: Lint chart
+        run: helm lint --strict helm/trino-gateway
+      - name: Create kind cluster
+        uses: helm/kind-action@v1.10.0
+        with:
+          cluster_name: tgw
+      - name: load images
+        run: kind load docker-image --name tgw trino-gateway:10-SNAPSHOT-amd64 && kind load docker-image --name tgw trino-gateway:10-SNAPSHOT-arm64
```
Comment (mosabua) [inline .github/workflows/ci.yml:55]: how do we update the version here all the time? Can we make it generic to the current build somehow?

Comment (mosabua) [reply .github/workflows/ci.yml:55]: Or use some sort of variable

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -37,3 +37,24 @@ jobs:
...
+      - name: Run chart-testing (install)
+        run: helm install tgw helm/trino-gateway  --set "image.repository=trino-gateway" --set "image.tag=$(cat ./trino-gateway-version.txt | tr -d '\n')"  --set "image.tag=10-SNAPSHOT-amd64" --set "limits.memory=100Mi" --set "limits.cpu=0"
+      - name: wait for deployment
+        run: kubectl rollout status deployment tgw-trino-gateway
```
Comment (mosabua) [inline .github/workflows/ci.yml:59]: tgw-trino-gateway .. maybe just tgw or trino-gateway or something simple

File: helm/trino-gateway/values.yaml [hunk 3]
```diff
@@ -4,7 +4,7 @@ image:
     repository: "trinodb/trino-gateway"
     pullPolicy: IfNotPresent
     # Override the image tag whose default is the chart appVersion.
-    # tag: "8-SNAPSHOT"
+    #tag: "9-SNAPSHOT"
```
Comment (mosabua) [inline helm/trino-gateway/values.yaml:7]: 10 ?

File: helm/trino-gateway/values.yaml [hunk 4]
```diff
@@ -37,6 +37,12 @@ config:
     serverConfig:
         node.environment: test
         http-server.http.port: 8080
+    dataStore:
+        #This stores the URLs of backend Trino servers and query history
```
Comment (mosabua) [inline helm/trino-gateway/values.yaml:41]: Maybe

```suggestion
        # The connection details for the backend database for Trino Gateway and Trino query history
```

or just

        # The connection details for the backend database for Trino Gateway

**Review by mosabua — COMMENTED** [review]
A few nits. Also would love it if @nineinchnick could take a peek

**Comment by oneonestar** [comment]
Thanks for the fix.
Could you split the fix and the addition of the CI test into two commits?

File: .github/workflows/ci.yml (hunk 2 above)
Comment (mosabua) [reply .github/workflows/ci.yml:59]: yeah .. I think we should go with test or so instead and also clarify where that comes from

**Review by mosabua — APPROVED** [review]
One minor comment .. but essentially good to go. I think we probably also need to update some docs as follow up, but we can do that as the general clean up before 10.

**Pull request merged**